codalab/worker/worker.py

```diff
@@ -584,44 +583,6 @@ def process_runs(self):
                 if run_state.stage != RunStage.FINISHED
             }
 
-    def assign_cpu_and_gpu_sets(self, request_cpus, request_gpus):
-        """
-        Propose a cpuset and gpuset to a bundle based on given requested resources.
-        Note: no side effects (this is important: we don't want to maintain more state than necessary)
-
-        Arguments:
-            request_cpus: integer
-            request_gpus: integer
-
-        Returns a 2-tuple:
-            cpuset: assigned cpuset (str indices).
-            gpuset: assigned gpuset (str indices).
-
-        Throws an exception if unsuccessful.
-        """
-        cpuset, gpuset = set(map(str, self.cpuset)), set(map(str, self.gpuset))
-
-        for run_state in self.runs.values():
-            if run_state.stage == RunStage.RUNNING:
-                cpuset -= run_state.cpuset
-                gpuset -= run_state.gpuset
-
-        if len(cpuset) < request_cpus:
-            raise Exception(
-                "Requested more CPUs (%d) than available (%d currently out of %d on the machine)"
-                % (request_cpus, len(cpuset), len(self.cpuset))
-            )
-        if len(gpuset) < request_gpus:
-            raise Exception(
-                "Requested more GPUs (%d) than available (%d currently out of %d on the machine)"
-                % (request_gpus, len(gpuset), len(self.gpuset))
-            )
-
-        def propose_set(resource_set, request_count):
```

Comment:
Is there a chance that the user will actually get _less_ CPUs / GPUs than they request? If so, then we might need to keep this function.